tests/crm: Marvell specific changes

sureshsekar28 opened the pull request:
As per Marvell design nexthop increments only after adding route
Signed-off-by: sureshsekar28 <sureshsekar@marvell.com>

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/sonic-net/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
As per Marvell design nexthop increments only after adding route

Summary:
Fixes # As per Marvell design nexthop increments only after adding route

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ x] Test case(new/improvement)


### Back port request
- [x ] 201911
- [ x] 202012
- [x ] 202205

### Approach
#### What is the motivation for this PR?
As per Marvell design nexthop increments only after adding route
#### How did you do it?
script enhancement
#### How did you verify/test it?
script passing with the change
#### Any platform specific information?
Marvell specific changes
#### Supported testbed topology if it's a new test case?

### Documentation
NA

Blueve commented:
Do we need this fix to 201911 and 202012?

sureshsekar28 commented:
Yes

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Jing Kan ***@***.***>
Sent: Thursday, December 8, 2022 7:21:28 PM
To: sonic-net/sonic-mgmt ***@***.***>
Cc: Suresh Sekar ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [sonic-net/sonic-mgmt] tests/crm: Marvell specific changes (PR #6994)

External Email
________________________________

Do we need this fix to 201911 and 202012?

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sonic-2Dnet_sonic-2Dmgmt_pull_6994-23issuecomment-2D1342769050&d=DwMCaQ&c=nKjWec2b6R0mOyPaz7xtfQ&r=B-jTQXldORQCDMAUB6mwEwDb_1YTzZkwzSjZuec-4Y4&m=KM-cgUiWoiTjtqrE3F04l6qZR2hIKnw1lrljidQZGh3GiJXw2tUOzL-j1XXcRQEJ&s=A9WVFQQv2aEWHQggOV0NkUN8Y22tkjtLoJLXklVZeZU&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AYWTPY7IRN2M5IJOEQDNQE3WMHRWBANCNFSM6AAAAAASXYZFHA&d=DwMCaQ&c=nKjWec2b6R0mOyPaz7xtfQ&r=B-jTQXldORQCDMAUB6mwEwDb_1YTzZkwzSjZuec-4Y4&m=KM-cgUiWoiTjtqrE3F04l6qZR2hIKnw1lrljidQZGh3GiJXw2tUOzL-j1XXcRQEJ&s=npIoPO-wCoOVp50hUNV5F4Vqud0ahPd4nQyX-9crhG8&e=>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

Blueve commented on tests/crm/test_crm.py:
```diff
@@ -552,29 +552,49 @@ def test_crm_route(duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_fro
+        vlan_mem_del_cmd = "config vlan member del 1000 Ethernet1"
+        vlan_mem_add_cmd =  "config vlan member add 1000 Ethernet1 -u"
+        if ip_ver == "4":
+           ptf_ip_add_cmd = "ifconfig eth1 {}/24 up".format(nexthop)
```
How about use existing ptf method:
https://github.com/sonic-net/sonic-mgmt/blob/fec265e539769efa2eab2222cdf719f994116dee/tests/common/devices/ptf.py#L51
and 
https://github.com/sonic-net/sonic-mgmt/blob/fec265e539769efa2eab2222cdf719f994116dee/tests/common/devices/ptf.py#L90

sureshsekar28 replied:
i dont find add_ip_to_dev method,
![image](https://user-images.githubusercontent.com/103626723/206656529-993326b0-2aeb-49d8-99ce-5b3d4a91dac7.png)

sureshsekar28 replied:
Thanks, addressed review comments

Blueve commented on tests/crm/test_crm.py:
```diff
@@ -552,29 +552,49 @@ def test_crm_route(duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_fro
+           ip_remove_cmd = "config interface ip remove Ethernet1 2001::2/64"
+           nexthop_add_cmd = "config route add prefix 3001::0/64 nexthop {}".format(nexthop)
+           nexthop_del_cmd = "config route del prefix 3001::0/64 nexthop {}".format(nexthop)
+        asichost.shell(vlan_mem_del_cmd)
```
How about use sonic host method?
https://github.com/sonic-net/sonic-mgmt/blob/fec265e539769efa2eab2222cdf719f994116dee/tests/common/devices/sonic.py#L2078
https://github.com/sonic-net/sonic-mgmt/blob/fec265e539769efa2eab2222cdf719f994116dee/tests/common/devices/sonic.py#L2068

sureshsekar28 replied:
I dont find del_member_from_vlan method,
(Pdb) asichost.del_member_from_vlan(1000,'Ethernet1')
*** AttributeError: 'SonicAsic' object has no attribute 'del_member_from_vlan'
(Pdb) dir(asichost)
['_MULTI_ASIC_DOCKER_NAME', '_MULTI_ASIC_SERVICE_NAME', '__class__', '__delattr__', '__dict__', '__doc__', '__format__', '__getattribute__', '__hash__', '__init__', '__module__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'asic_index', 'bgp_drop_rule', 'bgp_facts', 'check_bgp_statistic', 'cli_ns_option', 'command', 'config_facts', 'config_ip_intf', 'config_portchannel', 'config_portchannel_member', 'create_ssh_tunnel_sai_rpc', 'delete_container', 'get_active_ip_interfaces', 'get_asic_namespace', 'get_bgp_statistic', 'get_critical_services', 'get_default_route_from_app_db', 'get_docker_cmd', 'get_docker_name', 'get_extended_minigraph_facts', 'get_ip_route_info', 'get_portchannel_and_members_in_ns', 'get_queue_oid', 'get_router_mac', 'get_service_name', 'interface_facts', 'ip_cmd', 'is_backend_portchannel', 'is_container_running', 'is_default_route_removed_from_app_db', 'is_it_backend', 'is_it_frontend', 'is_service_running', 'namespace', 'ns_arg', 'os_version', 'ping_v4', 'port_exists', 'port_on_asic', 'portchannel_on_asic', 'ports', 'queue_oid', 'remove_ssh_tunnel_sai_rpc', 'reset_service', 'restart_service', 'run_redis_cli_cmd', 'run_redis_cmd', 'run_sonic_db_cli_cmd', 'run_vtysh', 'shell', 'show_interface', 'show_ip_interface', 'shutdown_interface', 'sonic_db_cli', 'sonic_release', 'sonichost', 'start_service', 'startup_interface', 'stop_service', 'switch_arptable']

sureshsekar28 replied:
Thanks,addressed review comments

Blueve commented:
Hi @sureshsekar28, there are some new styling issues introduced by your change, can you help fix them?
```
tests/crm/test_crm.py:555:121: E501 line too long (141 > 120 characters)
tests/crm/test_crm.py:561:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:561:40: E231 missing whitespace after ','
tests/crm/test_crm.py:562:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:562:45: E231 missing whitespace after ','
tests/crm/test_crm.py:563:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:564:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:565:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:566:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:568:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:568:40: E231 missing whitespace after ','
tests/crm/test_crm.py:569:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:569:45: E231 missing whitespace after ','
tests/crm/test_crm.py:570:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:571:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:572:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:573:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:574:53: E231 missing whitespace after ','
tests/crm/test_crm.py:574:53: E231 missing whitespace after ','
tests/crm/test_crm.py:610:51: E231 missing whitespace after ','
tests/crm/test_crm.py:610:63: E231 missing whitespace after ','
```

Blueve reviewed (changes requested):
Hi @sureshsekar28, there are some new styling issues introduced by your change, can you help fix them?
```
tests/crm/test_crm.py:555:121: E501 line too long (141 > 120 characters)
tests/crm/test_crm.py:561:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:561:40: E231 missing whitespace after ','
tests/crm/test_crm.py:562:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:562:45: E231 missing whitespace after ','
tests/crm/test_crm.py:563:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:564:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:565:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:566:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:568:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:568:40: E231 missing whitespace after ','
tests/crm/test_crm.py:569:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:569:45: E231 missing whitespace after ','
tests/crm/test_crm.py:570:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:571:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:572:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:573:12: E111 indentation is not a multiple of 4
tests/crm/test_crm.py:574:53: E231 missing whitespace after ','
tests/crm/test_crm.py:574:53: E231 missing whitespace after ','
tests/crm/test_crm.py:610:51: E231 missing whitespace after ','
tests/crm/test_crm.py:610:63: E231 missing whitespace after ','
```

sureshsekar28 commented:
Thanks. I have corrected styling issues. Please check now.

From: Jing Kan ***@***.***>
Date: Monday, 12 December 2022 at 2:01 PM
To: sonic-net/sonic-mgmt ***@***.***>
Cc: Suresh Sekar ***@***.***>, Mention ***@***.***>
Subject: [EXT] Re: [sonic-net/sonic-mgmt] tests/crm: Marvell specific changes (PR #6994)
External Email
________________________________

@Blueve requested changes on this pull request.

Hi @sureshsekar28<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sureshsekar28&d=DwMCaQ&c=nKjWec2b6R0mOyPaz7xtfQ&r=B-jTQXldORQCDMAUB6mwEwDb_1YTzZkwzSjZuec-4Y4&m=h5kuoOLnPo8IjK72yGDR77-GBmqIP8vuCHV5Vv0P2FoCKqzcD2cB-5E2mdlvCZQ_&s=51fdelZPX1e7xIFA3Ql8AkllMJ3yuGrbS-4FnowMy94&e=>, there are some new styling issues introduced by your change, can you help fix them?

tests/crm/test_crm.py:555:121: E501 line too long (141 > 120 characters)

tests/crm/test_crm.py:561:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:561:40: E231 missing whitespace after ','

tests/crm/test_crm.py:562:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:562:45: E231 missing whitespace after ','

tests/crm/test_crm.py:563:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:564:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:565:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:566:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:568:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:568:40: E231 missing whitespace after ','

tests/crm/test_crm.py:569:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:569:45: E231 missing whitespace after ','

tests/crm/test_crm.py:570:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:571:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:572:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:573:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:574:53: E231 missing whitespace after ','

tests/crm/test_crm.py:574:53: E231 missing whitespace after ','

tests/crm/test_crm.py:610:51: E231 missing whitespace after ','

tests/crm/test_crm.py:610:63: E231 missing whitespace after ','

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sonic-2Dnet_sonic-2Dmgmt_pull_6994-23pullrequestreview-2D1213084118&d=DwMCaQ&c=nKjWec2b6R0mOyPaz7xtfQ&r=B-jTQXldORQCDMAUB6mwEwDb_1YTzZkwzSjZuec-4Y4&m=h5kuoOLnPo8IjK72yGDR77-GBmqIP8vuCHV5Vv0P2FoCKqzcD2cB-5E2mdlvCZQ_&s=dzbVLjclq3kSKw4m7fZIZTZfYvfhwhOvZ1fZgGBgPzg&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AYWTPY7RWUZIGBGDZK22OPDWM3PF7ANCNFSM6AAAAAASXYZFHA&d=DwMCaQ&c=nKjWec2b6R0mOyPaz7xtfQ&r=B-jTQXldORQCDMAUB6mwEwDb_1YTzZkwzSjZuec-4Y4&m=h5kuoOLnPo8IjK72yGDR77-GBmqIP8vuCHV5Vv0P2FoCKqzcD2cB-5E2mdlvCZQ_&s=wmc4Xg_psWIaDpgKuTElQjIU0dfgrjiWotW6T52yMIk&e=>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

sureshsekar28 commented:
I have addressed all the review comments. Please review.

Thanks,
Suresh.

From: Suresh Sekar ***@***.***>
Date: Monday, 12 December 2022 at 3:07 PM
To: sonic-net/sonic-mgmt ***@***.***>, sonic-net/sonic-mgmt ***@***.***>
Cc: Mention ***@***.***>
Subject: Re: [EXT] Re: [sonic-net/sonic-mgmt] tests/crm: Marvell specific changes (PR #6994)
Thanks. I have corrected styling issues. Please check now.

From: Jing Kan ***@***.***>
Date: Monday, 12 December 2022 at 2:01 PM
To: sonic-net/sonic-mgmt ***@***.***>
Cc: Suresh Sekar ***@***.***>, Mention ***@***.***>
Subject: [EXT] Re: [sonic-net/sonic-mgmt] tests/crm: Marvell specific changes (PR #6994)
External Email
________________________________

@Blueve requested changes on this pull request.

Hi @sureshsekar28<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sureshsekar28&d=DwMCaQ&c=nKjWec2b6R0mOyPaz7xtfQ&r=B-jTQXldORQCDMAUB6mwEwDb_1YTzZkwzSjZuec-4Y4&m=h5kuoOLnPo8IjK72yGDR77-GBmqIP8vuCHV5Vv0P2FoCKqzcD2cB-5E2mdlvCZQ_&s=51fdelZPX1e7xIFA3Ql8AkllMJ3yuGrbS-4FnowMy94&e=>, there are some new styling issues introduced by your change, can you help fix them?

tests/crm/test_crm.py:555:121: E501 line too long (141 > 120 characters)

tests/crm/test_crm.py:561:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:561:40: E231 missing whitespace after ','

tests/crm/test_crm.py:562:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:562:45: E231 missing whitespace after ','

tests/crm/test_crm.py:563:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:564:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:565:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:566:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:568:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:568:40: E231 missing whitespace after ','

tests/crm/test_crm.py:569:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:569:45: E231 missing whitespace after ','

tests/crm/test_crm.py:570:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:571:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:572:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:573:12: E111 indentation is not a multiple of 4

tests/crm/test_crm.py:574:53: E231 missing whitespace after ','

tests/crm/test_crm.py:574:53: E231 missing whitespace after ','

tests/crm/test_crm.py:610:51: E231 missing whitespace after ','

tests/crm/test_crm.py:610:63: E231 missing whitespace after ','

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sonic-2Dnet_sonic-2Dmgmt_pull_6994-23pullrequestreview-2D1213084118&d=DwMCaQ&c=nKjWec2b6R0mOyPaz7xtfQ&r=B-jTQXldORQCDMAUB6mwEwDb_1YTzZkwzSjZuec-4Y4&m=h5kuoOLnPo8IjK72yGDR77-GBmqIP8vuCHV5Vv0P2FoCKqzcD2cB-5E2mdlvCZQ_&s=dzbVLjclq3kSKw4m7fZIZTZfYvfhwhOvZ1fZgGBgPzg&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AYWTPY7RWUZIGBGDZK22OPDWM3PF7ANCNFSM6AAAAAASXYZFHA&d=DwMCaQ&c=nKjWec2b6R0mOyPaz7xtfQ&r=B-jTQXldORQCDMAUB6mwEwDb_1YTzZkwzSjZuec-4Y4&m=h5kuoOLnPo8IjK72yGDR77-GBmqIP8vuCHV5Vv0P2FoCKqzcD2cB-5E2mdlvCZQ_&s=wmc4Xg_psWIaDpgKuTElQjIU0dfgrjiWotW6T52yMIk&e=>.
You are receiving this because you were mentioned.Message ID: ***@***.***>
